src/ansys/mapdl/core/misc.py

```diff
@@ -425,3 +429,644 @@ def check_valid_start_instance(start_instance):
+
+    def __repr__(self):
+        if not self._stats:  # pragma: no cover
+            self._update()
+
+        return "\n".join(
+            [
+                f"{each_name}:".ljust(25) + f"{getattr(self, each_attr)}".ljust(25)
+                for each_name, each_attr in self._repr_keys.items()
+            ]
+        )
+
+    @property
+    @update_information_first(False)
+    def product(self):
+        """Retrieve the product from the MAPDL instance."""
+        return self._get_product()
+
+    @product.setter
+    def product(self, value):
+        """Set the value for the MAPDL Product
+
+        .. warning:: By default, you cannot change 'product' property.
+
+        """
+        raise ValueError("The property 'product' cannot be changed.")
+
+    @property
+    @update_information_first(False)
+    def mapdl_version(self):
+        """Retrieve the MAPDL version from the MAPDL instance."""
+        return self._get_mapdl_version()
+
+    @mapdl_version.setter
+    def mapdl_version(self, value):
+        """Set the value for the MAPDL Version
+
+        .. warning:: By default, you cannot change 'mapdl_version' property.
+
+        """
+        raise ValueError("The property 'mapdl_version' cannot be changed")
+
+    @property
+    @update_information_first(False)
+    def mapdl_version_release(self):
+        """Retrieve the MAPDL version release from the MAPDL instance."""
+        st = self._get_mapdl_version()
+        return self._get_between("RELEASE", "BUILD", st).strip()
+
+    @mapdl_version_release.setter
+    def mapdl_version_release(self, value):
+        """Set the value for the MAPDL version release
+
+        .. warning:: By default, you cannot change 'mapdl_version_release' property.
+
+        """
+        raise ValueError("The property 'mapdl_version_release' cannot be changed")
+
+    @property
+    @update_information_first(False)
+    def mapdl_version_build(self):
+        """Retrieve the MAPDL version build from the MAPDL instance."""
+        st = self._get_mapdl_version()
+        return self._get_between("BUILD", "UPDATE", st).strip()
+
+    @mapdl_version_build.setter
+    def mapdl_version_build(self, value):
+        """Set the value for the MAPDL Mapdl Version Build
+
+        .. warning:: By default, you cannot change 'mapdl_version_build' property.
+
+        """
+        raise ValueError("The property 'mapdl_version_build' cannot be changed")
+
+    @property
+    @update_information_first(False)
+    def mapdl_version_update(self):
+        """Retrieve the MAPDL version update from the MAPDL instance."""
+        st = self._get_mapdl_version()
+        return self._get_between("UPDATE", "", st).strip()
+
+    @mapdl_version_update.setter
+    def mapdl_version_update(self, value):
+        """Set the value for the MAPDL Mapdl Version Update
+
+        .. warning:: By default, you cannot change 'mapdl_version_update' property.
+
+        """
```

Comment:
Setters won't be rendered in our docs, you can delete all setter docstrings.

Reply:
Don